exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestVarlenDecimal.java

```diff
@@ -139,8 +145,32 @@ public void testWriteReadCsv() throws Exception {
           .baselineValues(new BigDecimal(bigDecimalValue))
           .go();
     } finally {
-      resetSessionOption(ExecConstants.OUTPUT_FORMAT_OPTION);
-      test("drop table if exists dfs.tmp.%s", tableName);
+      client.resetSession(ExecConstants.OUTPUT_FORMAT_OPTION);
+      run("drop table if exists dfs.tmp.%s", tableName);
+    }
+  }
+
+  @Test
+  public void testUnionAllWithDifferentScales() throws Exception {
+    try {
+      run("create table dfs.tmp.t as select cast(999999999999999 as decimal(15,0)) as d");
+
+      String query = "select cast(1000 as decimal(10,1)) as d\n" +
+          "union all \n" +
+          "select 596.000 as d \n" +
+          "union all \n" +
+          "select d from dfs.tmp.t";
+
+      testBuilder()
+          .sqlQuery(query)
+          .unOrdered()
+          .baselineColumns("d")
+          .baselineValues(new BigDecimal("1000.000"))
+          .baselineValues(new BigDecimal("596.000"))
+          .baselineValues(new BigDecimal("999999999999999.000"))
```

Comment:
This tests *values*. Would it make sense to the use the `RowSet` version so you can also verify *types*, which is the focus of this PR?

Reply:
Not sure why this will be complex.
Here is an example:
```
    RowSet actual = client.queryBuilder().sql(sql).rowSet();

    TupleMetadata expectedSchema = new SchemaBuilder()
        // add your schema here 
        .buildSchema();
    RowSet expected = new RowSetBuilder(client.allocator(), expectedSchema)
       // add your columns here
        .build();
    RowSetUtilities.verify(expected, actual);
```

Reply:
@arina-ielchiieva, thanks for pointing to the example, initially I tried a similar approach, but union operator returns multiple batches, so in the case of using `client.queryBuilder().sql(sql).rowSet()`, it will fail with (`IllegalStateException`)[https://github.com/apache/drill/blob/09b805aea4dafe50555b23945302cf8f6c491de8/exec/java-exec/src/test/java/org/apache/drill/test/QueryBuilder.java#L359]. I also tried to use `rowSetIterator()`, as it was advised in the error message, but it returns `QueryRowSetIterator` which actually caused complications for me. I tried several approaches, one of them is iterating through it and comparing separate row sets, which corresponds to returned batches, but such a check would look obscure since the first batch was empty.
So I ended up with the simpler check for schema using `testBuilder()`.

Reply:
Thanks for explanation.